src/utils.ts

```diff
@@ -58,7 +58,13 @@ export function parseDateFields(input: JsonValue, depth = 0): ReturnJsonValue {
     return Object.entries(input).reduce((output, [k, v]) => {
         const isValObject = !!v && typeof v === 'object';
         if (k.endsWith(PARSE_DATE_FIELDS_KEY_SUFFIX)) {
-            output[k] = v ? new Date(v as string) : v;
+            if (v) {
+                const d = new Date(v as string);
+                // eslint-disable-next-line no-self-compare
+                output[k] = d.getTime() === d.getTime() ? d : v as string;
```

Comment:
maybe better this way?

```suggestion
                output[k] = isNaN(d.getTime()) ? v as string : d;
```